kfs-core/src/main/java/edu/arizona/kfs/fp/document/web/struts/ProcurementCardForm.java

```diff
@@ -29,6 +36,41 @@ public ProcurementCardForm() {
     public String getDocuwareTableParam() {
         return SpringContext.getBean(ParameterService.class).getParameterValueAsString(ProcurementCardDocument.class, KFSConstants.DOCUWARE_TABLE_PARAMETER);
     }
+    
+    @Override
+    public List<ExtraButton> getExtraButtons() {
+        extraButtons.clear();
+
+        //Needed to refresh workflow doc to get the state and approval flag set correctly
+        getProcurementCardDocument().getDocumentHeader().getWorkflowDocument().refresh();
+
+        List<RouteNodeInstance> routeNodeInstances = getProcurementCardDocument().getDocumentHeader().getWorkflowDocument().getCurrentRouteNodeInstances();
+        String node = routeNodeInstances.get(0).getName();
+        
+        //Show RETURN to Reconciler button only if user is an approver (and not reconciler)
+        if (getProcurementCardDocument().getDocumentHeader().getWorkflowDocument().isEnroute() &&
+                getProcurementCardDocument().getDocumentHeader().getWorkflowDocument().isApprovalRequested() &&
+                !node.equalsIgnoreCase(PROCUREMENT_CARD_RECONCILER)) {
+            extraButtons.add(returnToReconcilerButton());
```

Comment:
Please make getReturnToReconcilerButton() a lazy instantiator for the ExtraButton that will return every time the same instance, since the button has very static parameters.